src/main/java/seedu/address/logic/parser/job/FindJobParser.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.parser.job;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_JOBID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_JOBTITLE;
+
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.job.FindJob;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.job.Job;
+
+public class FindJobParser implements Parser<FindJob> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindJob
+     * and returns a FindJob object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindJob parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_JOBTITLE, PREFIX_JOBID);
+        Predicate<Job> predicate;
+
+        if (!anyPrefixesPresent(argMultimap, PREFIX_JOBTITLE, PREFIX_JOBID) || !argMultimap.getPreamble().isEmpty()) {
```

Comment:
Might want to add a check for both prefix present? E.g 
`|| arePrefixesPresent(argMultimap, PREFIX_JOBTITLE, PREFIX_JOBID)`

Since if both prefixes are present, then the parser will only use the first prefix which is JobTitle.